### PR TITLE
refactor: use the `t` defined in the module

### PR DESCRIPTION
**Pull request by ven-k**
`t` is now available in three flavours. Thus, allow a downstream package to import the appropriate one, and use that to define variables in the mtk-models.

## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
tests fail?

**Review by ChrisRackauckas — APPROVED** [review]

File: test/runtests.jl [hunk 1]
```diff
@@ -94,5 +94,3 @@ end
         @safetestset "BifurcationKit Extension Test" include("extensions/bifurcationkit.jl")
     end
 end
-
-@safetestset "Model Parsing Test" include("model_parsing.jl")
```
Comment (ven-k) [inline test/runtests.jl:98]: Removed as this is duplicate of this test: https://github.com/SciML/ModelingToolkit.jl/pull/2569/files#diff-3b9314a6f9f2d7eec1d0ef69fa76cfabafdbe6d0df923768f9ec32f27a249c63R47

**Comment by ven-k** [comment]
Model Parsing Tests pass: https://github.com/SciML/ModelingToolkit.jl/actions/runs/8506330803/job/23296377185?pr=2569#step:6:1422

**Comment by ven-k** [comment]
While here, I've improved tests for constants (and it's metadata) in mtkmodel.

----

@ChrisRackauckas the model parsing tests are fixed.
https://github.com/SciML/ModelingToolkit.jl/actions/runs/8512837408/job/23315296803#step:6:1302

**Pull request merged**